### PR TITLE
Add support for `fetch-remote-resources` and http proxies

### PR DESCRIPTION
**Pull request by alexjfisher**

### CODE REVIEW

**Comment by treydock** [comment]
+1

**Comment by ares** [comment]
since it's listed as WIP, what's missing or planned to be added?

**Comment by treydock** [comment]
I've patched my local systems with the `--fetch-remote-resources` portion of this PR and it's working nicely.

**Comment by alexjfisher** [comment]
@ares Sorry, I got distracted and moved onto something else.  IIRC we spoke at FOSDEM.  There's a change needed in the foreman_scap_client puppet module (apologies to @treydock for not opening my own PR with changes for that), but it wasn't necessary to make changes in the foreman plugin itself to use this new functionality.  This PR should be good to go (and I had it all working in a PoC environment, but haven't got round to rolling out openscap/policies yet).

**Comment by ares** [comment]
I see, thanks for update @alexjfisher. @xprazak2 told me he plans to review this. Btw there's already a http://projects.theforeman.org/issues/21116 opened in our redmine so I assigned the issue there to you. It might be a good idea to change commit message to "Fixes #21116 - ..." so it's linked properly.

**Comment by ares** [comment]
And I think it also solves http://projects.theforeman.org/issues/21390 since one can use HTTP proxy, great!

**Comment by alexjfisher** [comment]
Can I link to both by referencing the issues in the commit message body, (or does it have to be in the title)?

**Comment by ares** [comment]
I think it needs to be the title (first line)

File: lib/foreman_scap_client/client.rb [hunk 1]
```diff
@@ -62,7 +82,12 @@ def scan_command
       else
         profile = ''
       end
-      "oscap xccdf eval #{profile} #{tailoring_subcommand} --results-arf #{results_path} #{config[@policy_id][:content_path]}"
+      fetch_remote_resources = if config[:fetch_remote_resources]
```
Comment (xprazak2) [inline lib/foreman_scap_client/client.rb:85]: You can make this shorter by changing it into:

```ruby
fetch_remote_resources = '--fetch-remote-resources' if config[:fetch_remote_resources]
```

File: lib/foreman_scap_client/client.rb [hunk 2]
```diff
@@ -38,16 +39,35 @@ def load_config
...
+        puts error_str
         exit(2)
       end
     end
 
+    def scan_command_env_vars
+      if http_proxy_uri
+        {
+          'HTTP_PROXY'  => http_proxy_uri,
+          'HTTPS_PROXY' => http_proxy_uri
```
Comment (xprazak2) [inline lib/foreman_scap_client/client.rb:57]: Is having `http_proxy_uri` as HTTPS_PROXY intentional?

**Comment by xprazak2** [comment]
Nice work. I apologize it took me so long to test, I was occupied elsewhere.

@alexjfisher, I am not sure how the proxy env variables are supposed to be used, could you give me an example how to test them?

@alexjfisher, @treydock, could you sync up whether [43](https://github.com/theforeman/puppet-foreman_scap_client/pull/43) or [45](https://github.com/theforeman/puppet-foreman_scap_client/pull/45) for the puppet module should eventually go in? They are essentially identical and they both work with this.

File: lib/foreman_scap_client/client.rb (hunk 2 above)
Comment (alexjfisher) [reply lib/foreman_scap_client/client.rb:57]: It was.  I'm setting both environment variables as I don't know whether the remote resources will be http or https.  I use squid for both http and https connections.  If I was downloading something with wget, I might set environment variables as follows.
```
HTTP_PROXY=http://squid.example.com:3128 wget http://insecure_url.example.com/foo
HTTPS_PROXY=http://squid.example.com:3128 wget https://secure_url.example.com/foo
```

In both case, the proxy_uri is identical.

**Comment by xprazak2** [comment]
Merged as [8565e3cc9d8452aa6c5c412e3ccccd9e028d5c7](https://github.com/theforeman/foreman_scap_client/commit/f8565e3cc9d8452aa6c5c412e3ccccd9e028d5c7), thanks!

I slightly updated commit message to link it with Redmine issues.

**Pull request closed without merging**